packages/shared/src/variants/baduk_utils.ts

```diff
@@ -49,13 +49,17 @@ export function isLegacyBadukConfig(
   );
 }
 
-export function mapToNewConfig(config: LegacyBadukConfig): NewBadukConfig {
+export function mapToNewConfig<ConfigT extends LegacyBadukConfig>(
```

Comment:
Hm is there a reason against using a simpler signature?

```
export function mapToNewConfig(
  config: LegacyBadukConfig,
): LegacyBadukConfig & NewGridBadukConfig {
  return {
    ...config,
    board: {
      type: BoardPattern.Grid,
      width: config.width,
      height: config.height,
    },
  };
```

Technically speaking I think the type error thrown by transpiler is correct. If I write `mapToNewConfig<LegacyBadukConfig & {someNewProp: string}>(config)` then the function signature does not check out.

Reply:
I think I was trying to capture the following:

```
const config = {
      komi: 6.5,
      width: 19,
      height: 19,
      xShift: 0,
      yShift: 1,
}
const sanitizedConfig = mapToNewConfig(config);
sanitizedConfig.xShift;
                ^^^^^^
Property 'xShift' does not exist on type 'NewGridBadukConfig'.
```

But it occurs to me that `DriftGoConfig` inherits from `NewGridBadukConfig`, so we don't really make use of this complex typing.   I'll go ahead and remove it

Reply:
Ah in that case maybe the "fix" is to change the type of the input config to ConfigT:
```
export function mapToNewConfig<ConfigT extends LegacyBadukConfig>(
  config: ConfigT,
): ConfigT & NewGridBadukConfig {
  return {
    ...config,
    board: {
      type: BoardPattern.Grid,
      width: config.width,
      height: config.height,
    },
  };
```

Reply:
Of course!  I totally missed that haha

Fixed in latest commit